tests/rados/pool_tests.py

```diff
@@ -620,97 +620,250 @@ def run(ceph_cluster, **kw):
+        5. POOL_TOO_MANY_PGS
+        """
         log.debug(
             "Test to verify the warnings generated by autoscaler for PG count on pools"
         )
-        pool_configs = config["verify_autoscaler_warn"]["pool_configs"]
-        pool_configs_path = config["verify_autoscaler_warn"]["pool_configs_path"]
+        pool_configs = config["verify_pool_warnings"]["pool_configs"]
+        pool_configs_path = config["verify_pool_warnings"]["pool_configs_path"]
 
         try:
             with open(pool_configs_path, "r") as fd:
                 pool_conf_file = yaml.safe_load(fd)
 
-            # Changing the default autoscaler value to warn for newly created pools
-            rados_obj.configure_pg_autoscaler(default_mode="warn")
-
             pools = []
             for i in pool_configs:
+                # Changing the default autoscaler value to warn for newly created pools
+                rados_obj.configure_pg_autoscaler(default_mode="warn")
+
                 pool = pool_conf_file[i["type"]][i["conf"]]
                 pool["pg_num"] = 1
+                if rhbuild.split("-")[0] in ["7.1"]:
+                    pool["app_name"] = None
                 create_given_pool(rados_obj, pool)
-                rados_obj.autoscaler_pool_settings(
-                    pg_autoscale_mode="warn", pool_name=pool["pool_name"]
-                )
                 pools.append(pool["pool_name"])
-            log.info(f"Created {len(pools)} pools for testing. pools : {pools}")
-            log.debug(
-                f"autoscale status on pools: \n{rados_obj.run_ceph_command(cmd='ceph osd pool autoscale-status')}"
-            )
+                time.sleep(10)
 
-            # Starting to write data continuously onto the two pools to trigger the warning
-            # refer: https://bugzilla.redhat.com/show_bug.cgi?id=2172795#c24
-            for pname in pools:
-                rados_obj.bench_write(pool_name=pname, byte_size="1Kb", max_objs=500)
+                log.debug(
+                    f"autoscale status on pools: \n{rados_obj.node.shell(['ceph osd pool autoscale-status'])}"
+                )
 
-            time.sleep(20)
-            if not rados_obj.check_health_warning(warning="POOL_TOO_FEW_PGS"):
-                # Starting to write data continuously onto the two pools and checking for generation of warning
-                if not check_pg_warning(rados_obj, pools):
-                    log.error("PG count warning not generated for the pools")
-                    raise Exception("No warning generated by PG Autoscaler")
+                if rhbuild.split("-")[0] in ["7.1"]:
+                    log.debug("Bug fixed only in 7.1.")
+                    # Scenario 1: Generate POOL_APP_NOT_ENABLED
+                    log.debug(
+                        "Checking if the warning for not enabling application on pool is generated"
+                    )
+                    POOL_APP_NOT_ENABLED = False
```

Comment:
The warning gets generated fairly easily and consistently if any IOs is started on the pool, would be like to run bench or put before checking the warning?

Reply:
Writing of IOs is not needed, discussed offline,